Sources/GRPCCodeGen/Internal/Translator/IDLToStructuredSwiftTranslator.swift

```diff
@@ -109,21 +117,23 @@ extension IDLToStructuredSwiftTranslator {
     if let noNamespaceServices = servicesByNamespace[""] {
       let namespaces = servicesByNamespace.keys
       for service in noNamespaceServices {
-        if namespaces.contains(service.name) {
+        if namespaces.contains(service.generatedName) {
           throw CodeGenError(
             code: .nonUniqueServiceName,
             message: """
-              Services with no namespace must not have the same names as the namespaces. \
-              \(service.name) is used as a name for a service with no namespace and a namespace.
+              Services with no namespace must not have the same generated names as the namespaces. \
+              \(service.generatedName) is used as a generated name for a service with no namespace and a namespace.
               """
           )
         }
       }
     }
 
-    // Check that service names are unique within each namespace.
+    // Check that service names and service generated names are unique within each namespace.
```

Comment:
We only call this function with the descriptors grouped by the generated namespace, why do we still need to check the non-generated service names?

I think the only modifications you need to make to your existing checks should be to use the generated names instead of the current ones _and_ to add one more check which ensures that all descriptors are unique.

Reply:
So, for the methods, I think it is important to check that the upper case names are unique within a service (because they are enum names inside the service enum), but also that the lowerCase names are unique within a service, as they will be used for function declaration in the same protocols. So, in this case I think it's useful to have 3 checks (the third one is for the descriptor).  But for the services I will only check the upperCase names and the descriptors (that they are unique).